connectors/sources/sharepoint_online.py

```diff
@@ -647,105 +666,128 @@ async def get_docs(self, filtering=None):
+                        ):
+                            drive_item["_id"] = drive_item["id"]
+                            drive_item["object_type"] = "drive_item"
+                            drive_item["_timestamp"] = drive_item[
+                                "lastModifiedDateTime"
+                            ]
+
+                            download_func = None
+
+                            if "@microsoft.graph.downloadUrl" in drive_item:
+                                modified_date = datetime.strptime(
+                                    drive_item["lastModifiedDateTime"],
+                                    "%Y-%m-%dT%H:%M:%SZ",
                                 )
-
-                        yield drive_item, download_func
-
-                async for site_list in self.client.site_lists(site["id"]):
-                    site_list["_id"] = site_list["id"]
-                    site_list["object_type"] = "site_list"
-
-                    yield site_list, None
-
-                    async for list_item in self.client.site_list_items(
-                        site["id"], site_list["id"]
-                    ):
-                        list_item["_id"] = list_item["id"]
-                        list_item["object_type"] = "list_item"
-                        content_type = list_item["contentType"]["name"]
-
-                        if content_type in [
-                            "Web Template Extensions",
-                            "Client Side Component Manifests",
-                        ]:  # TODO: make it more flexible. For now I ignore them cause they 404 all the time
-                            continue
-
-                        if "Attachments" in list_item["fields"]:
-                            async for list_item_attachment in self.client.site_list_item_attachments(
-                                site["webUrl"], site_list["name"], list_item["id"]
-                            ):
-                                list_item_attachment["_id"] = list_item_attachment[
-                                    "odata.id"
-                                ]
-                                list_item_attachment[
-                                    "object_type"
-                                ] = "list_item_attachment"
-                                list_item_attachment["_timestamp"] = list_item[
-                                    "lastModifiedDateTime"
-                                ]
-                                attachment_download_func = partial(
-                                    self.get_attachment_content, list_item_attachment
+                                if (
+                                    max_data_age
+                                    and modified_date
+                                    < datetime.utcnow()
+                                    - timedelta(seconds=max_data_age)
+                                ):
+                                    logger.warning(
```

Comment:
I expect that we will have lots of those. do we want to lower the level to debug?

Reply:
It is very noisy, I will lower it to debug